### PR TITLE
add tests for dump/restore of multiple databases

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/4998/

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ Here's an example of dumping data from a non-standard endpoint, using a dedicate
 
     arangodump --server.endpoint tcp://192.168.173.13:8531 --server.username backup --server.database mydb --output-directory "dump"
 
+In contrast to the above call `--server.database` must not be specified when duming all databases using `--all-databases true`:
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]: `true` is not needed for binary options, but probably better to be explicit so 👍

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 2]
```diff
@@ -58,6 +58,11 @@ To create the target database whe restoring, use a command like this:
 
     arangorestore --server.username backup --server.database newdb --create-database true --input-directory "dump"
 
+In contrast to the above calls, when working with multiple databases using `--all-databases true` the parameter `--server.database mydb` must not be specified:
+
+    arangorestore --server.username backup --all-databases true --create-database true --input-directory "dump-multiple"
+    
+
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]: Extra blank line not needed
```suggestion
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 3]
```diff
@@ -15,16 +15,22 @@ directory, you can use the parameter *--overwrite true* to confirm this:
     arangodump --output-directory "dump" --overwrite true
 
 _arangodump_ will by default connect to the *_system* database using the default
-endpoint. If you want to connect to a different database or a different endpoint,
-or use authentication, you can use the following command-line options:
+endpoint. To override the endpoint, or specify a different user, use one of the
+following commandline parameters:
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]: It's actually not just command line parameters, they could also be set via a config file. A neutral word would be _startup options_, see https://docs.arangodb.com/3.4/Manual/Administration/Configuration/

```suggestion
following startup options:
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 4]
```diff
@@ -15,21 +15,31 @@ directory, you can use the parameter *--overwrite true* to confirm this:
...
 
-- `--server.database <string>`: name of the database to connect to
 - `--server.endpoint <string>`: endpoint to connect to
 - `--server.username <string>`: username
 - `--server.password <string>`: password to use (omit this and you'll be prompted for the
   password)
 - `--server.authentication <bool>`: whether or not to use authentication
 
+If you want to connect to a different database or dump all databases you can additionaly
+use the following command-line options (please note that the specified user must have access
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]:
```suggestion
use the following startup options:
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 5]
```diff
@@ -15,21 +15,31 @@ directory, you can use the parameter *--overwrite true* to confirm this:
...
+to the databases):
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]:
```suggestion
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 6]
```diff
@@ -15,21 +15,31 @@ directory, you can use the parameter *--overwrite true* to confirm this:
...
+
+- `--all-databases true`: must have access to all databases, and not specify a database.
+- `--server.database <string>`: name of the database to connect to
+
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]:
```suggestion

Note that the specified user must have access to the databases.

```

Perhaps make this a hint box even? Also, what happens the user has access to some, but not all databases? Does it fail in the middle of the export? Or are permissions checked for all DBs before dumping starts?

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 7]
```diff
@@ -23,16 +23,22 @@ documents found in the input directory *dump*. Please note that the input direct
 must have been created by running *arangodump* before.
 
 _arangorestore_ will by default connect to the *_system* database using the default
-endpoint. If you want to connect to a different database or a different endpoint,
-or use authentication, you can use the following command-line options:
+endpoint. To override the endpoint, or specify a different user, use one of the
+following commandline parameters:
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]:
```suggestion
following startup options:
```

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 8]
```diff
@@ -23,16 +23,22 @@ documents found in the input directory *dump*. Please note that the input direct
...
 
-- `--server.database <string>`: name of the database to connect to
 - `--server.endpoint <string>`: endpoint to connect to
 - `--server.username <string>`: username
 - `--server.password <string>`: password to use (omit this and you'll be prompted for the
   password)
 - `--server.authentication <bool>`: whether or not to use authentication
 
+If you want to connect to a different database or dump all databases you can additionaly
+use the following command-line options (please note that the specified user must have access
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]:
```suggestion
use the following startup options:
```

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 9]
```diff
@@ -23,16 +23,22 @@ documents found in the input directory *dump*. Please note that the input direct
...
+to the databases):
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]:
```suggestion
```

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 10]
```diff
@@ -23,16 +23,22 @@ documents found in the input directory *dump*. Please note that the input direct
...
+
+- `--all-databases true`: must have access to all databases, and not specify a database.
+- `--server.database <string>`: name of the database to connect to
+
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]:
```suggestion

 Note that the specified user must have access to the database(s).
 
```

File: Documentation/Books/Manual/Programs/Arangorestore/Examples.md [hunk 11]
```diff
@@ -58,6 +64,10 @@ To create the target database whe restoring, use a command like this:
 
     arangorestore --server.username backup --server.database newdb --create-database true --input-directory "dump"
 
+In contrast to the above calls, when working with multiple databases using `--all-databases true` the parameter `--server.database mydb` must not be specified:
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangorestore/Examples.md]:
```suggestion
In contrast to the above calls, when working with multiple databases using `--all-databases true`
the parameter `--server.database mydb` must not be specified:
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 12]
```diff
@@ -15,21 +15,31 @@ directory, you can use the parameter *--overwrite true* to confirm this:
...
 - `--server.username <string>`: username
 - `--server.password <string>`: password to use (omit this and you'll be prompted for the
   password)
 - `--server.authentication <bool>`: whether or not to use authentication
 
+If you want to connect to a different database or dump all databases you can additionaly
+use the following command-line options (please note that the specified user must have access
+to the databases):
+
+- `--all-databases true`: must have access to all databases, and not specify a database.
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]: Be more explicit maybe?

```suggestion
- `--all-databases true`: user must have access to all databases, and not specify a database.
```

File: Documentation/Books/Manual/Programs/Arangodump/Examples.md [hunk 13]
```diff
@@ -15,21 +15,31 @@ directory, you can use the parameter *--overwrite true* to confirm this:
...
+- `--server.database <string>`: name of the database to connect to
+
 Here's an example of dumping data from a non-standard endpoint, using a dedicated
 [database name](../../Appendix/Glossary.md#database-name):
 
     arangodump --server.endpoint tcp://192.168.173.13:8531 --server.username backup --server.database mydb --output-directory "dump"
 
+In contrast to the above call `--server.database` must not be specified when duming all databases using `--all-databases true`:
```
Comment (Simran-B) [inline Documentation/Books/Manual/Programs/Arangodump/Examples.md]:
```suggestion
In contrast to the above call `--server.database` must not be specified when dumping
all databases using `--all-databases true`:
```

**Pull request merged**